gpii/node_modules/solutionsRegistry/src/js/SolutionsUtils.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * GPII Solutions Utils - Independent helper functions and utilities for working with
+ * solutions and their schemas.
+ *
+ * Copyright 2019 Raising the Floor - US
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013)
+ * under grant agreement no. 289016.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.universal.solutionsRegistry");
+
+/**
+ * This function will take a schema from a supported setting in the solutions
+ * registry and return metadata on it's default value, including the type of
+ * the default value. The function will always return a block of json in order
+ * to avoid any ambiguities for a situation where the default value is perhaps
+ * `null`. A key `hasDefault` will be `true` or `false` to communicate this.
+ * If the schema does have a default value, then the properties `type` and
+ * `default` can also be included. Note that there may not always be a `type`
+ * that can be included, the inclusion of this in the return if optional and
+ * will be included if a `type` is declared.
+ *
+ * This is NOT meant to be an exhaustive coverage of all the possible ways
+ * default values can be set for any JSON Schema. We merely want to cover the
+ * cases in use in the solutions registry, focussing on single primitive values,
+ * and situations where we next one level deep due to a requirement of having a
```

Comment:
Please fix "where we next one level deep... "

Reply:
This is fixed up now.